src/pkgcheck/checks/codingstyle.py

```diff
@@ -167,31 +167,95 @@ def short_desc(self):
         return f"{self.variable} unnecessary slash strip on line{_pl(self.lines)}: {lines}"
 
 
+class DoublePrefixInPath(base.Error):
+    """Ebuild uses two consecutive paths including EPREFIX.
+
+    Ebuild combines two path variables (or a variable and a getter), both
+    of which include EPREFIX, resulting in double prefixing.  This is the case
+    when combining many pkg-config-based or alike getters with ED or EROOT.
+
+    For example, '${ED}$(python_get_sitedir)' should be replaced
+    with '${D}$(python_get_sitedir)'.
+    """
+
+    __slots__ = ("category", "package", "version", "variable", "lines")
+    threshold = base.versioned_feed
+
+    def __init__(self, pkg, variable, lines):
+        super().__init__()
+        self._store_cpv(pkg)
+        self.variable = variable
+        self.lines = tuple(lines)
+
+    @property
+    def short_desc(self):
+        lines = ', '.join(map(str, self.lines))
+        return (f"{self.variable} concatenates two variables containing "
+                f"EPREFIX on line{_pl(self.lines)}: {lines}")
+
+
 class PathVariablesCheck(base.Template):
     """Scan ebuild for path variables with various issues."""
 
     feed_type = base.ebuild_feed
-    known_results = (MissingSlash, UnnecessarySlashStrip)
-    variables = ('ROOT', 'EROOT', 'D', 'ED')
+    known_results = (MissingSlash, UnnecessarySlashStrip, DoublePrefixInPath)
+    prefixed_variables = ('EROOT', 'ED')
+    variables = ('ROOT', 'D') + prefixed_variables
+    prefixed_getters = (
```

Comment:
Would it be possible to flag these in a formatted way somehow using the eclass doc markup so we don't have to maintain the list manually in pkgcheck once eclass doc parsing support is written?

Reply:
Hmm, I don't see a problem with that. I suppose an extra tag for VARIABLE and FUNCTION thingies.